### PR TITLE
Specialcase jdk.vm.ci

### LINKED ISSUE #929: jdk.vm.ci. not found
This is in a branch with some updates to the BOM via MaximumLinkageErrors test. Looks like something we might want to special case. Need to research further. 

Class jdk.vm.ci.meta.ResolvedJavaMethod is not found
Class jdk.vm.ci.services.Services is not found
Class jdk.vm.ci.meta.ResolvedJavaType is not found
Class jdk.vm.ci.meta.Signature is not found
Class jdk.vm.ci.meta.JavaType is not found
Class jdk.vm.ci.meta.JavaConstant is not found
Class jdk.vm.ci.meta.JavaKind is not found
Class jdk.vm.ci.code.TargetDescription is not found
Class jdk.vm.ci.amd64.AMD64$Flag is not found
Class jdk.vm.ci.aarch64.AArch64 is not found
Class jdk.vm.ci.code.Architecture is not found
Class jdk.vm.ci.meta.JavaKind$FormatWithToString is not found
Class jdk.vm.ci.meta.ResolvedJavaField is not found
Class jdk.vm.ci.meta.VMConstant is not found
Class jdk.vm.ci.meta.Constant is not found
Class jdk.vm.ci.meta.ConstantReflectionProvider is not found
Class jdk.vm.ci.meta.Assumptions$AssumptionResult is not found
Class jdk.vm.ci.meta.PlatformKind is not found
Class jdk.vm.ci.code.CodeUtil is not found
Class jdk.vm.ci.meta.MetaAccessProvider is not found
Class jdk.vm.ci.amd64.AMD64$CPUFeature is not found
Class jdk.vm.ci.amd64.AMD64 is not found
Class jdk.vm.ci.code.stack.StackIntrospection is not found
Class jdk.vm.ci.runtime.JVMCI is not found
Class jdk.vm.ci.runtime.JVMCIRuntime is not found
Class jdk.vm.ci.runtime.JVMCIBackend is not found
Class jdk.vm.ci.code.stack.InspectedFrame is not found
Class jdk.vm.ci.code.stack.InspectedFrameVisitor is not found
Class jdk.vm.ci.code.site.Infopoint is not found
Class jdk.vm.ci.code.site.InfopointReason is not found
Class jdk.vm.ci.meta.SpeculationLog$SpeculationReason is not found
Class jdk.vm.ci.meta.SpeculationLog is not found
Class jdk.vm.ci.meta.SpeculationLog$Speculation is not found
Class jdk.vm.ci.meta.DeoptimizationAction is not found
Class jdk.vm.ci.meta.DeoptimizationReason is not found
Class jdk.vm.ci.code.VirtualObject is not found
Class jdk.vm.ci.code.StackSlot is not found
Class jdk.vm.ci.code.DebugInfo is not found
Class jdk.vm.ci.code.BytecodeFrame is not found
Class jdk.vm.ci.meta.JavaValue is not found
Class jdk.vm.ci.code.ValueUtil is not found
Class jdk.vm.ci.meta.AllocatableValue is not found
Class jdk.vm.ci.code.site.Call is not found
Class jdk.vm.ci.code.site.ExceptionHandler is not found
Class jdk.vm.ci.meta.LocalVariableTable is not found
Class jdk.vm.ci.meta.Local is not found
Class jdk.vm.ci.aarch64.AArch64$CPUFeature is not found
Class jdk.vm.ci.meta.Value is not found
Class jdk.vm.ci.common.JVMCIError is not found
Class jdk.vm.ci.code.Register is not found
Class jdk.vm.ci.code.ReferenceMap is not found
Class jdk.vm.ci.code.site.DataPatch is not found
Class jdk.vm.ci.code.site.DataSectionReference is not found
Class jdk.vm.ci.code.site.ConstantReference is not found
Class jdk.vm.ci.meta.MemoryAccessProvider is not found
Class jdk.vm.ci.code.RegisterConfig is not found
Class jdk.vm.ci.code.CodeCacheProvider is not found
Class jdk.vm.ci.code.CallingConvention is not found
Class jdk.vm.ci.code.ValueKindFactory is not found
Class jdk.vm.ci.code.CallingConvention$Type is not found
Class jdk.vm.ci.meta.ValueKind is not found
Class jdk.vm.ci.meta.SerializableConstant is not found
Class jdk.vm.ci.code.CompiledCode is not found
Class jdk.vm.ci.code.site.Reference is not found
Class jdk.vm.ci.aarch64.AArch64Kind is not found
Class jdk.vm.ci.hotspot.aarch64.AArch64HotSpotRegisterConfig is not found
Class jdk.vm.ci.code.RegisterArray is not found
Class jdk.vm.ci.code.RegisterAttributes is not found
Class jdk.vm.ci.code.RegisterValue is not found
Class jdk.vm.ci.aarch64.AArch64$Flag is not found
Class jdk.vm.ci.hotspot.HotSpotJVMCIRuntime is not found
Class jdk.vm.ci.meta.JavaField is not found
Class jdk.vm.ci.meta.JavaMethod is not found
Class jdk.vm.ci.meta.ExceptionHandler is not found
Class jdk.vm.ci.meta.UnresolvedJavaType is not found
Class jdk.vm.ci.meta.ConstantPool is not found
Class jdk.vm.ci.common.NativeImageReinitialize is not found
Class jdk.vm.ci.meta.MetaUtil is not found
Class jdk.vm.ci.meta.JavaTypeProfile$ProfiledType is not found
Class jdk.vm.ci.meta.JavaTypeProfile is not found
Class jdk.vm.ci.meta.TriState is not found
Class jdk.vm.ci.meta.ResolvedJavaMethod$Parameter is not found
Class jdk.vm.ci.meta.LineNumberTable is not found
Class jdk.vm.ci.meta.JavaMethodProfile is not found
Class jdk.vm.ci.meta.JavaMethodProfile$ProfiledMethod is not found
Class jdk.vm.ci.hotspot.HotSpotResolvedJavaMethod is not found
Class jdk.vm.ci.hotspot.HotSpotResolvedJavaField is not found
Class jdk.vm.ci.hotspot.HotSpotResolvedJavaType is not found
Class jdk.vm.ci.hotspot.HotSpotSignature is not found
Class jdk.vm.ci.meta.DefaultProfilingInfo is not found
Class jdk.vm.ci.meta.PrimitiveConstant is not found
Class jdk.vm.ci.code.InstalledCode is not found
Class jdk.vm.ci.code.InvalidInstalledCodeException is not found
Class jdk.vm.ci.amd64.AMD64Kind is not found
Class jdk.vm.ci.code.Register$RegisterCategory is not found
Class jdk.vm.ci.meta.ProfilingInfo is not found
Class jdk.vm.ci.hotspot.HotSpotJVMCICompilerFactory$CompilationLevelAdjustment is not found
Class jdk.vm.ci.hotspot.HotSpotSpeculationLog is not found
Class jdk.vm.ci.code.BytecodePosition is not found
Class jdk.vm.ci.runtime.JVMCICompiler is not found
Class jdk.vm.ci.code.BailoutException is not found
Class jdk.vm.ci.meta.Assumptions is not found
Class jdk.vm.ci.meta.InvokeTarget is not found
Class jdk.vm.ci.code.CodeUtil$DefaultRefMapFormatter is not found
Class jdk.vm.ci.code.site.Mark is not found
Class jdk.vm.ci.code.StackLockValue is not found
Class jdk.vm.ci.code.CompilationRequest is not found
Class jdk.vm.ci.meta.RawConstant is not found
Class jdk.vm.ci.meta.MethodHandleAccessProvider is not found
Class jdk.vm.ci.meta.MethodHandleAccessProvider$IntrinsicMethod is not found
Class jdk.vm.ci.code.RegisterSaveLayout is not found
Class jdk.vm.ci.sparc.SPARC is not found
Class jdk.vm.ci.sparc.SPARC$CPUFeature is not found
Class jdk.vm.ci.sparc.SPARCKind is not found
Class jdk.vm.ci.hotspot.HotSpotVMConfigAccess is not found
Class jdk.vm.ci.hotspot.HotSpotResolvedObjectType is not found
Class jdk.vm.ci.hotspot.HotSpotInstalledCode is not found
Class jdk.vm.ci.hotspot.HotSpotCompilationRequestResult is not found
Class jdk.vm.ci.hotspot.HotSpotCompilationRequest is not found
Class jdk.vm.ci.services.JVMCIServiceLocator is not found
Class jdk.vm.ci.runtime.JVMCICompilerFactory is not found
Class jdk.vm.ci.hotspot.HotSpotVMEventListener is not found
Class jdk.vm.ci.common.InitTimer is not found
Class jdk.vm.ci.hotspot.HotSpotCallingConventionType is not found
Class jdk.vm.ci.hotspot.HotSpotCodeCacheProvider is not found
Class jdk.vm.ci.hotspot.HotSpotCompiledCode$Comment is not found
Class jdk.vm.ci.hotspot.HotSpotCompiledNmethod is not found
Class jdk.vm.ci.hotspot.HotSpotCompiledCode is not found
Class jdk.vm.ci.code.site.Site is not found
Class jdk.vm.ci.hotspot.HotSpotVMConfigStore is not found
Class jdk.vm.ci.hotspot.HotSpotJVMCICompilerFactory is not found
Class jdk.vm.ci.hotspot.HotSpotJVMCICompilerFactory$CompilationLevel is not found
Class jdk.vm.ci.hotspot.HotSpotConstant is not found
Class jdk.vm.ci.hotspot.HotSpotCompressedNullConstant is not found
Class jdk.vm.ci.code.CompilationRequestResult is not found
Class jdk.vm.ci.hotspot.HotSpotForeignCallTarget is not found
Class jdk.vm.ci.code.Location is not found
Class jdk.vm.ci.hotspot.HotSpotReferenceMap is not found
Class jdk.vm.ci.hotspot.HotSpotMetaspaceConstant is not found
Class jdk.vm.ci.hotspot.HotSpotObjectConstant is not found
Class jdk.vm.ci.hotspot.HotSpotConstantReflectionProvider is not found
Class jdk.vm.ci.hotspot.HotSpotConstantPool is not found
Class jdk.vm.ci.hotspot.HotSpotResolvedPrimitiveType is not found
Class jdk.vm.ci.hotspot.HotSpotMemoryAccessProvider is not found
Class jdk.vm.ci.hotspot.HotSpotMetaAccessProvider is not found
Class jdk.vm.ci.hotspot.HotSpotSentinelConstant is not found
Class jdk.vm.ci.meta.Assumptions$Assumption is not found
Class jdk.vm.ci.hotspot.HotSpotNmethod is not found

### PR DESCRIPTION
**Pull request by suztomo**
Fixes #929 

@elharo Although we discussed about checking the source to "jdk.vm.ci" packages, LinkageChecker does not know Maven artifacts. It only know class names. This would make the specialcasing a bit complex if we check the sources. And thus this PR only checks the missing symbol package.

### CODE REVIEW

**Comment by suztomo** [comment]
We can check the file name of the source. Let me update.

**Comment by suztomo** [comment]
Also checking package names.

File: boms/cloud-oss-bom/pom.xml [hunk 1]
```diff
@@ -46,7 +46,7 @@
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
     <guava.version>28.1-android</guava.version>
     <google.cloud.java.version>0.102.0-alpha</google.cloud.java.version>
-    <io.grpc.version>1.22.1</io.grpc.version>
+    <io.grpc.version>1.23.0</io.grpc.version>
```
Comment (suztomo) [inline boms/cloud-oss-bom/pom.xml]: I will revert this once I confirm this specialcasing works.

**Comment by suztomo** [comment]
@elharo PTAL.

a70ae92 passed all presubmit checks with new gRPC version. Reverted gRPC version to clean PR.

File: dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/LinkageChecker.java [hunk 2]
```diff
@@ -160,12 +161,35 @@ private LinkageChecker(
...
+    SetMultimap<SymbolProblem, ClassFile> filteredMap =
+        Multimaps.filterEntries(problemToClass.build(), LinkageChecker::problemFilter);
     return ImmutableSetMultimap.copyOf(filteredMap);
   }
 
+  /**
+   * Returns true if the linkage error {@code entry} should be reported. False if it should be
+   * suppressed.
+   */
+  private static boolean problemFilter(Map.Entry<SymbolProblem, ClassFile> entry) {
```
Comment (elharo) [inline dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/LinkageChecker.java]: We have two different special cases here, that use different algorithms, but the same approach should work in both. I wonder if we should have a more general solution that lets us use configuration to whitelist instances we've verified are not problems.

Comment (suztomo) [reply dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/LinkageChecker.java]: I like the idea of configuration. It’s going to be easy to explain special cases.

**Pull request merged**